File: dev-docs/snippets/workflows/terminate.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Terminate a running workflow.
export async function terminate(
  client: JsBaoClient,
  workflowKey: string,
  runKey: string,
) {
  // #region example
  const result = await client.workflows.terminate({ workflowKey, runKey });
  // #endregion example
  return result;
}

File: dev-docs/snippets/workflows/terminate.swift
import JsBaoClient

// Terminate a running workflow. Pass the same `contextDocId` the run was
// started with so the server can route to the right per-document instance.
func terminate(client: JsBaoClient, workflowKey: String, runKey: String) async throws {
  // #region example
  let result = try await client.workflows.terminate(
    workflowKey: workflowKey,
    runKey: runKey
  )
  // #endregion example
  _ = result
}
